index.js

```diff
@@ -2,23 +2,29 @@
 
 const Promise = require('bluebird');
 const readInput = Promise.promisify(require('read'));
-const argv = require('yargs').argv;
+const argv = require('yargs').array('domain').argv;
 const xtend = require('xtend');
 const getCertificate = require('./lib.js');
 
-const read = (options) => () => {
+const read = (options) => (forceReadInput = false) => {
```

Comment:
I've slightly modified how `yargs` handles the argument `domain` [here](https://github.com/monoblaine/gitlab-letsencrypt/blob/5cf96af5504d031d5ffd22c0f63aad14697da26a/index.js#L5):

```diff
- const argv = require('yargs').argv;
+ const argv = require('yargs').array('domain').argv;
```

Thanks to this change, we can have `yargs` parse the following line:

```
--domain example.com www.example.com
```

and give us the `domain` argument as an array. (no need to manually parse)

Therefore we cannot use the old implementation anymore. The old implementation checks whether the argument exists within the `argv` object as a `string`:

```js
const argument = typeof argv[options.argument] === 'string' ? argv[options.argument] : undefined
```

We no longer expect only strings, we may get an array. Therefore I didn't want to keep the old implementation and make it confusing. The implementation I suggest is something like this:

1. If we're not instructed to prompt user, try reading the value from `argv`.
2. Try validating the value: If the validation fails throw an error so the final `catch` is called and we're forced to skip `argv` and ask user to input value in the next iteration.